integration_tests/test_ibc.py

```diff
@@ -210,3 +211,89 @@ def test_ibc(cluster):
         assert [bal for bal in balances if int(bal["amount"]) > 0] == [
             {"amount": "10000000000", "denom": "basecro"},
         ]
+
+
+def test_update_chain_id(cluster):
+    data_root = next(iter(cluster.values())).data_root
+    # call chain-maind directly
+    raw = next(iter(cluster.values())).cosmos_cli().raw
+    relayer = ["relayer", "--home", data_root / "relayer"]
+
+    channels = json.loads(
+        raw(
+            "query",
+            "ibc",
+            "channel",
+            "channels",
+            node=cluster["ibc-1"].node_rpc(0),
+            output="json",
+        )
+    )["channels"]
+    assert channels[0]["state"] == "STATE_OPEN"
+    dst_channel = channels[0]["counterparty"]["channel_id"]
+    denom_hash = (
+        hashlib.sha256(f"transfer/{dst_channel}/basecro".encode()).hexdigest().upper()
+    )
+    # update chain-id
+    cluster["ibc-1"].stop_node()
+    genesis_data = json.loads(cluster["ibc-1"].export(0))
+    genesis_data["chain_id"] = "ibc-2"
+    upgrade_time = datetime.utcnow() + timedelta(seconds=5)
+    upgrade_time.replace(tzinfo=None).isoformat("T") + "Z"
+    genesis_data["genesis-time"] = str(upgrade_time)
+    cluster["ibc-1"].update_genesis(0, genesis_data)
+    cluster["ibc-1"].start_node(0)
+    rsp = json.loads(subprocess.check_output(relayer + ["chains", "list", "--json"]))
+    # the chain-id in relayer config is not changed
+    assert rsp == [
+        {
+            "key": "relayer",
+            "chain-id": "ibc-0",
+            "rpc-addr": "http://localhost:26657",
+            "account-prefix": "cro",
+            "gas-adjustment": 1.5,
+            "gas-prices": "0.0basecro",
+            "trusting-period": "336h",
+        },
+        {
+            "key": "relayer",
+            "chain-id": "ibc-1",
+            "rpc-addr": "http://localhost:26757",
+            "account-prefix": "cro",
+            "gas-adjustment": 1.5,
+            "gas-prices": "0.0basecro",
+            "trusting-period": "336h",
+        },
+    ]
+
+    # do a transfer from ibc-0 to ibc-2
+    balance_0 = cluster["ibc-0"].balance(cluster["ibc-0"].address("relayer"))
+    recipient = cluster["ibc-1"].address("relayer")
+    rsp = cluster["ibc-0"].ibc_transfer(
+        "relayer", recipient, "10000basecro", channels[0]["channel_id"], 1
+    )
+    assert rsp["code"] == 0, rsp["raw_log"]
+    time.sleep(10)
```

Comment:
Is it possible to detect the balance changed instead of using 10 secs sleep? We can set a 10 secs timeout to return test fail in case it really needs 10 secs to finish the transfer.